utilities/calibration-delta-updater/script/blf-calibration-delta-updater.py

```diff
@@ -35,69 +33,131 @@ def open_polydriver(robot_name, control_board, joints_list):
-    print_info("previous offset = " + str(['%.4f' % offset for offset in offsets]) + " deg")
-    print_info("new offset = " + str(['%.4f' % offset for offset in new_offsets]) + " deg")
+    group = parameters.get_group(robot_part)
+    if group is None:
+        raise RuntimeError('Unable to find the group named ' + robot_part)
 
+    joints = group.get_parameter_vector_string('joints_list')
+    board = group.get_parameter_string('control_board')
+    parameters.set_parameter_vector_string('joints_list', joints)
+    parameters.set_parameter_vector_string('remote_control_boards', [board])
+    values = group.get_parameter_vector_float('expected_values')
+
+    # populate the parameters handler with additional parameters required by the application. There is no need to
+    # expose these parameter in the configuration file
+    parameters.set_parameter_string('local_prefix', 'joint-offset-updater')
+    parameters.set_parameter_bool('check_for_nan', False)
+    parameters.set_parameter_bool('stream_joint_states', True)
+    return parameters, values, joints
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Automatically update the calibration delta of the robot '
+                                                 'configuration file.')
+    parser.add_argument('-i', '--input', type=str, required=True, help='Input xml file')
+    parser.add_argument('-o', '--output', type=str, required=True, help='Output xml file')
+    parser.add_argument('-p', '--part', type=str, required=True, help='Name of the part.')
+    parser.add_argument('--config', type=str, required=False, help='Path or name of the configuration file',
+                        default="blf-calibration-delta-updater-options.ini")
+    args = parser.parse_args()
+
+    # get the original offset
+    tree = etree.parse(args.input)
+    root = tree.getroot()
+    offsets = get_offsets(root)
+
+    # load the parameter handler
+    handler, expected_values, joints_list = parse_handler(args.config, args.part)
+
+    # Open the sensorbridge
+    control_board, sensor_bridge = open_polydriver(handler)
+
+    new_offsets = []
+    for i, joint in enumerate(joints_list):
+        key = ""
+        while key != 'y' and key != 'n':
+            print_info("Do you want to calibrate the joint named: " + joint + " [y|n]")
+            key = input()
+            if key == 'n':
+                new_offsets.append(0)
+            elif key == 'y':
+                print_info("Please move the joint in the expected configuration. Press enter when you are ready.")
+                input()
+                print(expected_values)
+                new_offsets.append(compute_joint_offset(sensor_bridge, i, expected_values))
+
+    updated_offset = np.array(offsets) + np.array(new_offsets)
+    print_info("Previous offsets = " + str(['%.4f' % offset for offset in offsets]) + " deg")
+    print_info("New offsets = " + str(['%.4f' % offset for offset in updated_offset]) + " deg")
```

Comment:
I wonder if it could make sense to add some deadzones. Usually, we tend to avoid changing the deltas if the difference is very small.

Reply:
We may ask the user if he wants to change the offset for a given joint. 🤔

Reply:
That's a good option. Otherwise, also something that sets a specific entry of ``new_offsets`` to zero if it is lower than a specified threshold could do the trick.